go.mod

```diff
@@ -1,11 +1,12 @@
 module github.com/pressly/goose/v3
 
-go 1.19
+go 1.21
```

Comment:
I'm not overly thrilled we have to bump the go directive because of a dependency. I know historically this field has been informational, but [starting with go1.21 this won't be the case](https://go.dev/doc/go1.21#tools).

> To improve forwards compatibility, Go 1.21 now reads the go line in a go.work or go.mod file as a strict minimum requirement: go 1.21.0 means that the workspace or module cannot be used with Go 1.20 or with Go 1.21rc1.

Running this command fails for me:

```
go mod tidy -compat=1.19 -go=1.19
```

There's an argument that the CLI code should be versioned and tracked separately from the library, so that we can use the latest drivers in `main.go` but keep goose as a package working with older versions.

This is the primary reason I haven't pulled the trigger. I'd like to think through this as to avoid backwards-compatible headaches with users not jumping on the latest version.

Reply:
I think there's a path forward here that aligns with something I've already been thinking about anyways. I need to decouple the CLI code + e2e tests from the goose library, i.e., that code should have its own go.{mod|sum} file tracking the drivers.

Since we build the goose binary with the latest supported Go version, this would offer enough flexibility to use the latest drivers.

Reply:
I see 🤔 

Would it be acceptable if we make min version 1.20 in libsql-client and so in goose?

Reply:
go1.20 might be defensible since Go itself has a release policy to support the last two major versions (go1.21 and go1.20).

This is also the most turbulent of times in Go because of the addition of generics, a lot of libraries are jumping on the latest and greatest features at the expense of backward compatibility.

I think I can do a release now, and another closer to go1.22 (release in Feb 2024). And then go1.20 will become the minimum supported.

TL;DR - I'd be okay having go1.20 live on `master` for a bit, since most projects will pin to the latest tagged anyways (which will be minimum go1.19). It just means if you want to use the Turso you'll have to pull `@master`